File: challenges/3-area-trapezio/js/ggonpereira/challenge-03.js
const prompt = require('prompt-sync')(); //importa prompt-sync para receber os dados do usuario

let baseMaior = parseInt(prompt("Digite a base maior do trapézio: "));
let baseMenor = parseInt(prompt("Digite a base menor do trapézio: "));
let altura = parseInt(prompt("Digite a altura do trapézio: "));

function calculaArea(bMaior, bMenor, alt) {
    let A = (bMaior + bMenor) * alt / 2;

    if (isNaN(A)) {
        return `Erro no cálculo, digite apenas números.`;
    } else {
        return `\nA área do trapézio com base menor de ${bMaior}, base maior de ${bMenor} e altura ${alt} é igual a: ${A}`;
    }
}

console.log(calculaArea(baseMaior, baseMenor, altura));

File: challenges/3-area-trapezio/python/tbonelis/challenge03.py
def bMaior(bMaior):
    while True:
        try:
            bMaior = float(input("Base maior: "))
        except (TypeError, ValueError):
            print("Por favor, digite um número.")
        else:
            return bMaior


def bMenor(bMenor):
    while True:
        try:
            bMenor = float(input("Base menor: "))
        except (TypeError, ValueError):
            print("Por favor, digite um número.")
        else:
            return bMenor


def altura(altura):
    while True:
        try:
            altura = float(input("Altura: "))
        except (TypeError, ValueError):
            print("Por favor, digite um número.")
        else:
            return altura


def areaT(bMaior, bMenor, altura):
    area = (bMaior + bMenor) * altura / 2
    return area

print(f"A área do trapézio é: {areaT(bMaior(bMaior), bMenor(bMenor), altura(altura))}")
